trl/trainer/sft_trainer.py

```diff
@@ -445,14 +442,14 @@ def make_inputs_require_grad(module, input, output):
     @wraps(Trainer.train)
     def train(self, *args, **kwargs):
         # Activate neftune right before training.
-        if self.neftune_noise_alpha is not None and not self._trainer_supports_neftune:
```

Comment:
It's weird, I read: if it doesn't support support netftune, then activate neftune

Reply:
Yes I am a bit confused too. There seems to be some redundant logic when alternating on checks between `neftune_noise_alpha` not being none and `self._trainer_supports_neftune` (which itself depends on `neftune_noise_alpha` earlier). I suppose there was a reason for it but I can't see it.